### PR TITLE
Unified update in

### PR DESCRIPTION
**Pull request by bsless**

### CODE REVIEW

File: src/clj_fast/inline.clj [hunk 1]
```diff
@@ -217,11 +217,37 @@
...
    (fn [m k v] `(c/assoc ~m ~k ~v))
    (fn [m k] `(c/get ~m ~k))
    m ksvs))
 
+(defmacro update-in->
+  "Like update-in but inlines the calls when a static sequence of keys is
+  provided.
+  Can take an unlimited number of [ks exprs] pairs.
+  Unlike update-in, expressions are not & args, i.e.:
+  (update-many-in m [:a :b] f [:a :b] (g 1))
```
Comment (yuhan0) [inline src/clj_fast/inline.clj]: Are both [:a :b] paths here supposed to be the same? 

Actually, the collapse step only allows one action per path, which is probably not what the user expects:
```clj
(update-in-> {:a {:b 1}}
  [:a :b] inc
  [:a :b] str)
;; => {:a {:b "1"}}
```
Maybe the preconditions should include `(distinct? (take-nth 2 ks-exprs))`

Comment (bsless) [reply src/clj_fast/inline.clj]: Nice catch on the docstring typo.
There's also no guarantee of order, so it's a good thing to add this precondition

**Comment by bsless** [comment]
@yuhan0 anything else you think needs to be addressed here?

**Comment by yuhan0** [comment]
looks good to me :)

File: src/clj_fast/inline.clj [hunk 2]
```diff
@@ -216,12 +216,40 @@
...
    (fn [m k v] `(c/assoc ~m ~k ~v))
    (fn [m k] `(c/get ~m ~k))
    m ksvs))
 
+(defmacro update-in->
+  "Like update-in but inlines the calls when a static sequence of keys is
+  provided.
+  Can take an unlimited number of [ks exprs] pairs.
+  Unlike update-in, expressions are not & args, i.e.:
+  (update-many-in m [:a :b] f [:a :c] (g 1))
```
Comment (yuhan0) [inline src/clj_fast/inline.clj]: Oh it looks like there's still a typo in the docstring here, update-many-in should be update-in->

**Pull request merged**